### PR TITLE
Fixed #21 -> Refactoring

### LINKED ISSUE #21: [ENH] Refactoring nested if structure
**Thx for the plugin!**

Some of the `if-if` statements can be summarized via `if + and`

Maybe that's something to consider

https://github.com/tiangolo/poetry-version-plugin/blob/89f9aaf02cbdc6a2a5972c07b167e78314ad153b/poetry_version_plugin/plugin.py#L57-L77

### PR DESCRIPTION
**Pull request by Anselmoo**
Refactoring `if-if` statements

Assignment Expressions via `:=`

### CODE REVIEW

**Review by dekoza — APPROVED** [review]

File: poetry_version_plugin/plugin.py [hunk 1]
```diff
@@ -26,8 +26,7 @@ def activate(self, poetry: Poetry, io: IO) -> None:
             io.write_error_line(message)
             raise RuntimeError(message)
         if version_source == "init":
-            packages = poetry.local_config.get("packages")
-            if packages:
+            if packages := poetry.local_config.get("packages"):
```
Comment (tiangolo) [inline poetry_version_plugin/plugin.py:29]: Thanks! But this would make it only compatible with Python 3.8, it would probably make sense deprecating below 3.8 but I want to first update it and then deprecate.

Anyway, deprecating would be better in another PR independent of the `if` refactors.

Could you revert this part?

**Review by tiangolo — CHANGES_REQUESTED** [review]

File: poetry_version_plugin/plugin.py (hunk 1 above)
Comment (Anselmoo) [reply poetry_version_plugin/plugin.py:29]: Will do

Comment (Anselmoo) [reply poetry_version_plugin/plugin.py]: image

Now it looks like that Python `version < 3.7 `has to happen earlier?

Comment (Anselmoo) [reply poetry_version_plugin/plugin.py]: > image
> Now it looks like that Python `version < 3.7 `has to happen earlier?

Due to rebase issue

Comment (Anselmoo) [reply poetry_version_plugin/plugin.py]: Fixed

**Comment by tiangolo** [comment]
Thanks for the patience with my reply! :sweat_smile: 

I just marked this project as deprecated, I'm currently not using it and I think these ideas can be achieved in better ways: https://github.com/tiangolo/poetry-version-plugin#-warning-deprecated-

Given that I'll close this one, but thanks for the interest! :coffee:

**Comment by Anselmoo** [comment]
@tiangolo, what kind of _**Python package manager**_ do you currently prefer?
